src/service.rs

```diff
@@ -407,19 +407,25 @@ impl RawServiceInfo {
         let account_password = to_wide(service_info.account_password.as_ref())
             .map_err(Error::InvalidAccountPassword)?;
 
-        // escape executable path and arguments and combine them into single command
-        let executable_path =
-            escape_wide(&service_info.executable_path).map_err(Error::InvalidExecutablePath)?;
-
+        // escape executable path and arguments and combine them into a single command
         let mut launch_command_buffer = WideString::new();
-        launch_command_buffer.push(executable_path);
+        if service_info.service_type.intersects(ServiceType::KERNEL_DRIVER | ServiceType::FILE_SYSTEM_DRIVER) {
+            // drivers do not support launch arguments
+            // also the path must not be quoted even if it contains spaces
+            let executable_path = WideCString::from_os_str(&service_info.executable_path).map_err(Error::InvalidExecutablePath)?;
+            launch_command_buffer.push(executable_path.to_ustring());
```

Comment:
Since arguments are not supported, I think we should probably return an error in this branch if `!service_info.launch_arguments.is_empty()`. Right? Because passing arguments is an invalidly formed `ServiceInfo` and silently just discarding them can lead to hard to debug situations for users of this crate. This would justify a new separate error variant.

Reply:
Changed, see a6e28b2.

Reply:
Great! Looks good to me.